server/routes/serviceProviderReferrals/reportingForm.test.ts

```diff
@@ -0,0 +1,154 @@
+import TestUtils from '../../../testutils/testUtils'
+import DateUtils from '../../utils/dateUtils'
+import ReportingForm from './reportingForm'
+
+describe(ReportingForm, () => {
+  describe('data', () => {
+    describe('with valid data', () => {
+      it('returns an object with two Calendar Day values', async () => {
+        const request = TestUtils.createRequest({
+          'from-date-year': '2021',
+          'from-date-month': '06',
+          'from-date-day': '12',
+          'to-date-year': '2021',
+          'to-date-month': '06',
+          'to-date-day': '13',
+        })
+
+        const result = await new ReportingForm(request).data()
+
+        expect(result.reportingDateRange!.fromIncludingDate).toEqual('2021-06-12')
+        expect(result.reportingDateRange!.toIncludingDate).toEqual('2021-06-13')
+        expect(result.error).toBeNull()
+      })
+    })
+
+    describe('with invalid data', () => {
+      describe('when the "from" date is after the "to" date', () => {
+        it('returns an error object with a message', async () => {
+          const request = TestUtils.createRequest({
+            'from-date-year': '2021',
+            'from-date-month': '06',
+            'from-date-day': '12',
+            'to-date-year': '2021',
+            'to-date-month': '06',
+            'to-date-day': '11',
+          })
+
+          const result = await new ReportingForm(request).data()
+
+          expect(result.reportingDateRange).toBeNull()
+          expect(result.error!.errors).toContainEqual({
```

Comment:
Is there a reason this isn't `toEqual` with a single-valued array? Maybe it would be good to check that we aren't getting other spurious errors here, only the one we expect. Ditto throughout this test file.